### ISSUE TITLE
Error when trying to launch test using contextual menu

### ISSUE THREAD

**Issue by NumberFourAG** [NumberFour/n4js#50]
IDEBUG-566
 "Unhandled event loop exception" in the Error Log

```
eu.numberfour.ide.runner.exceptions.InsolvableRuntimeEnvironmentException: Cannot resolve execution environment for the project eu.numberfour.mangelhaft.reporter.ide.test of type RUNTIME_LIBRARY.
    at eu.numberfour.ide.runner.RuntimeEnvironmentsHelper.findCompatibleRuntimeEnvironments(RuntimeEnvironmentsHelper.java:81)
    at eu.numberfour.ide.runner.ui.SupportingRunnerPropertyTester.findCompatibleRuntimeEnvironments(SupportingRunnerPropertyTester.java:181)
    at eu.numberfour.ide.runner.ui.SupportingRunnerPropertyTester.findCompatibleRuntimeEnvironments(SupportingRunnerPropertyTester.java:151)
    at eu.numberfour.ide.runner.ui.SupportingRunnerPropertyTester.test(SupportingRunnerPropertyTester.java:114)
    at org.eclipse.core.internal.expressions.Property.test(Property.java:58)
    at org.eclipse.core.internal.expressions.TestExpression.evaluate(TestExpression.java:100)
    at org.eclipse.core.internal.expressions.CompositeExpression.evaluateAnd(CompositeExpression.java:53)
    at org.eclipse.core.internal.expressions.AndExpression.evaluate(AndExpression.java:31)
    at org.eclipse.core.internal.expressions.CompositeExpression.evaluateAnd(CompositeExpression.java:53)
    at org.eclipse.core.internal.expressions.IterateExpression.evaluate(IterateExpression.java:160)
    at org.eclipse.core.internal.expressions.CompositeExpression.evaluateAnd(CompositeExpression.java:53)
    at org.eclipse.core.internal.expressions.WithExpression.evaluate(WithExpression.java:75)
    at org.eclipse.core.internal.expressions.CompositeExpression.evaluateAnd(CompositeExpression.java:53)
    at org.eclipse.core.internal.expressions.EnablementExpression.evaluate(EnablementExpression.java:55)
    at org.eclipse.debug.internal.ui.launchConfigurations.LaunchShortcutExtension.evalEnablementExpression(LaunchShortcutExtension.java:281)
    at org.eclipse.debug.ui.actions.ContextualLaunchAction.isApplicable(ContextualLaunchAction.java:287)
    at org.eclipse.debug.ui.actions.ContextualLaunchAction.fillMenu(ContextualLaunchAction.java:218)
    at org.eclipse.debug.ui.actions.ContextualLaunchAction$1.menuShown(ContextualLaunchAction.java:141)
    at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:255)
    at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
    at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4230)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1491)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1514)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1499)
    at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1299)
    at org.eclipse.jface.action.ActionContributionItem.copyMenu(ActionContributionItem.java:1276)
    at org.eclipse.jface.action.ActionContributionItem.handleShowProxy(ActionContributionItem.java:1256)
    at org.eclipse.jface.action.ActionContributionItem.access$4(ActionContributionItem.java:1245)
    at org.eclipse.jface.action.ActionContributionItem$8.handleEvent(ActionContributionItem.java:1218)
    at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
    at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4230)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1491)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1514)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1495)
    at org.eclipse.swt.widgets.Menu.menuWillOpen(Menu.java:811)
    at org.eclipse.swt.widgets.Display.windowProc(Display.java:5746)
    at org.eclipse.swt.internal.cocoa.OS.objc_msgSend(Native Method)
    at org.eclipse.swt.internal.cocoa.NSMenu.popUpContextMenu(NSMenu.java:77)
    at org.eclipse.swt.widgets.Menu._setVisible(Menu.java:280)
    at org.eclipse.swt.widgets.Display.runPopups(Display.java:4149)
    at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3691)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1127)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1018)
    at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:156)
    at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:654)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:337)
    at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:598)
    at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
    at eu.numberfour.ide.product.N4Application.start(N4Application.java:137)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1515)
```

Additional remarks:
- This problem occurs from the property tester of the context menu.
- Also one can experience this issue when trying to run tests from {{TestIDEReporter.n4js}} in the {{eu.numberfour.mangelhaft.reporter.ide.test}} project.
- This project is a {{RL}} project that has project dependencies.

**Comment by mor-n4** [comment]
probably related: #30

**Comment by mor-n4** [comment]
cannot reproduce

This issue seems outdated.
1. runtime libraries may no longer contain `.n4js` files (only `.n4jsd`).
2. project `eu.numberfour.mangelhaft.reporter.ide.test` mentioned in the issue description is no longer a runtime library (as stated in issue description) but now has project type `test`.
3. the code of class `SupportingRunnerPropertyTester` seems to have changed a lot since this ticket was created (when comparing the code to the information in the above stack trace).

I tried several things related to running ordinary code and tests in the context of runtime libraries and no such exception appeared.

Closing as "works for me".
